src/formatDate.js

```diff
@@ -50,7 +50,32 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const newSeparator = toFormat[3];
+  const newDate = date.split(oldSeparator);
+  const oldDateFormat = fromFormat.slice(0, -1);
+  const newDateFormat = toFormat.slice(0, -1);
+
+  if (fromFormat[0][0] !== toFormat[0][0]) {
+    newDate.reverse();
+    oldDateFormat.reverse();
+  }
+
```

Comment:
I have no idea what it does and how it works, but all I can say is that you really hacked the system👀
`  if (fromFormat[0][0] !== toFormat[0][0]) {
    newDate.reverse();
    oldDateFormat.reverse();
  }`

Reply:
> I have no idea what it does and how it works, but all I can say is that you really hacked the system👀 ` if (fromFormat[0][0] !== toFormat[0][0]) { newDate.reverse(); oldDateFormat.reverse(); }`

I compare the first characters of old format and new format and reverse old format( if it's should be reversed) , so they both fit in the 'for loop'. That is just how I have to live with my straightforward logic;

Reply:
As far as I can tell, you're doing the right thing, when life doesn't go your way, you make it to😂

Reply:
So, if fromFormat will be like yyyy/mm/dd, and toFormat will be like yyyy/dd/mm, your code won't be working properly then(( Reversing a date isn't the best solution here

Reply:
could you add this to the tests?